src/Microsoft.Health.Dicom.SqlServer/Features/ExtendedQueryTag/SqlExtendedQueryTagStoreV2.cs

```diff
@@ -134,29 +134,63 @@ internal static AddExtendedQueryTagsInputTableTypeV1Row ToAddExtendedQueryTagsIn
 
         public override async Task DeleteExtendedQueryTagAsync(string tagPath, string vr, CancellationToken cancellationToken = default)
         {
-            using (SqlConnectionWrapper sqlConnectionWrapper = await ConnectionWrapperFactory.ObtainSqlConnectionWrapperAsync(cancellationToken))
-            using (SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateSqlCommand())
-            {
-                V2.DeleteExtendedQueryTag.PopulateCommand(sqlCommandWrapper, tagPath, (byte)ExtendedQueryTagLimit.ExtendedQueryTagVRAndDataTypeMapping[vr]);
+            using SqlConnectionWrapper sqlConnectionWrapper = await ConnectionWrapperFactory.ObtainSqlConnectionWrapperAsync(cancellationToken);
+            using SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateSqlCommand();
 
-                try
-                {
-                    await sqlCommandWrapper.ExecuteNonQueryAsync(cancellationToken);
-                }
-                catch (SqlException ex)
+            sqlCommandWrapper.CommandText = @"
+SET NOCOUNT     ON
+SET XACT_ABORT  ON
+
+BEGIN TRANSACTION
+
+    DECLARE @tagKey INT
+
+    SELECT @tagKey = TagKey
+    FROM dbo.ExtendedQueryTag
+    WHERE dbo.ExtendedQueryTag.TagPath = @tagPath
+
+    -- Check existence
+    IF @@ROWCOUNT = 0
+        THROW 50404, 'extended query tag not found', 1
+
+    -- Delete index data
```

Comment:
I am not sure what other statuses are there, but wouldn't it be safer to only execute this if the state is "Deleting"?

Reply:
Not exactly. You should be able to delete in the middle of re-indexing (so you can cancel) and after re-indexing. In fact, it was a bit premature to add the `Deleting` status as the deletion operation is synchronous (users in theory should never see `Deleting` for now).